integration_tests/run_pyspark_from_build.sh

```diff
@@ -38,16 +38,55 @@ else
     then
         TEST_TAGS="-m $TEST_TAGS"
     fi
+    if [[ "${TEST_PARALLEL}" == "" ]];
+    then
+        # For now just assume that we are going to use the GPU on the system with the most memory
+        # and then divide it up into 3GiB chunks
+        # we subtract one for the main controlling process that will still
+        # launch an application, but it will never be used
+        TEST_PARALLEL=`nvidia-smi --query-gpu=memory.total --format=csv,noheader | awk '{if (MAX < $1){ MAX = $1}} END {print int(MAX / (2.3 * 1024)) - 1}'`
```

Comment:
This works for our pre-merge pipeline. Just curious do we plan to support parallelism on multiple GPUs?

Reply:
I don't have any plans right now. I am sure we could come up with a way.  The big issue is trying to tell different GPUs apart. I have 2 Nvidia GPUs in my desktop. One is for graphics and another is for compute. Cudf will not run on the graphics GPU at all so if I tried to launch anything on it, I would end up with problems. I also know that some people only have one graphics card that does both, and may not have all of the RAM available for compute. So I am thinking that we should switch this over from total memory to free memory to handle that case. If we can come up with a way to tell what the compute capability is for each GPU then we should be able to hack up how we launch pytest to do that. I'm just not sure how often we will run into that situation in practice.